### PR TITLE
[iOS][EP-836] rewrite card input form validation logic

### PR DESCRIPTION
**Pull request by AlexeiJovmir**
refactor card validator (remove unused code)
constants to card limits
unit tests for card validator servise

### CODE REVIEW

File: Source/Extensions/NSString/NSString+Additions.m [hunk 1]
```diff
@@ -42,78 +42,15 @@ - (NSString *)toCurrencySymbol {
...
-
-    if (network == CardNetworkAMEX && number.length > 15) {
-        if (error != NULL) {
-            *error = [NSError judoInputMismatchErrorWithMessage:@"check_card_number".localized];
-        }
-        return NO;
-    }
-
-    return YES;
+    return true;
```
Comment (mpetrenco) [inline Source/Extensions/NSString/NSString+Additions.m:53]: Rename to YES please

File: Source/Extensions/NSString/NSString+Additions.m [hunk 2]
```diff
@@ -42,77 +42,14 @@ - (NSString *)toCurrencySymbol {
...
-- (BOOL)isValidCardNumber:(NSString *)number
-               forNetwork:(CardNetwork)network
-         acceptedNetworks:(NSArray *)networks
-                    error:(NSError **)error {
-
-    if (number.length > 16 || ![number isNumeric]) {
-        if (error != NULL) {
-            *error = [NSError judoInputMismatchErrorWithMessage:@"check_card_number".localized];
-        }
+    if ([self rangeOfString:@"."].location != NSNotFound || !self.isLuhnValid) {
```
Comment (mpetrenco) [inline Source/Extensions/NSString/NSString+Additions.m:50]: what does `[self rangeOfString:@"."].location != NSNotFound` do ?

File: Source/Models/CardNetwork/JPCardNetwork.m [hunk 3]
```diff
@@ -58,80 +42,72 @@ + (instancetype)networkWith:(CardNetwork)type
...
-            [JPCardNetwork networkWith:CardNetworkMasterCard
-                        numberPrefixes:kMasterCardPrefixes],
-
-            [JPCardNetwork networkWith:CardNetworkDiscover
-                        numberPrefixes:kDiscoverPrefixes],
-        ];
-    });
-
-    return _networks;
++(BOOL)doesCardNumber:(NSString *)cardNumber matchRegex:(NSString *)regex {
```
Comment (mpetrenco) [inline Source/Models/CardNetwork/JPCardNetwork.m:51]: Please run `./clang-format.sh`

File: Source/Models/Constants/JPConstants.h [hunk 4]
```diff
@@ -28,26 +28,21 @@
...
+static NSString *const kChinaUnionPayPrefix = @"62";
```
Comment (mpetrenco) [inline Source/Models/Constants/JPConstants.h:45]: Do we need this, since we use RegEx ?

File: Source/Models/Constants/JPConstants.h [hunk 5]
```diff
@@ -71,4 +66,8 @@ static NSString *const kJudoErrorDomain = @"com.judo.error";
 
 static int const kMaximumLengthForConsumerReference = 40;
 
+static int const kJPMaxAMEXCardLength = 15;
```
Comment (mpetrenco) [inline Source/Models/Constants/JPConstants.h:69]: There's no reason to add `JP` for constant names.

File: Source/Models/Constants/JPConstants.h [hunk 6]
```diff
@@ -28,26 +28,19 @@
...
+static NSString *const kRegexVisa = @"^4\\d{0,15}";
+static NSString *const kRegexMasterCard = @"^(5[1-5]\\d{0,2}|22[2-9]\\d{0,1}|2[3-7]\\d{0,2})\\d{0,12}";
+static NSString *const kRegexMaestro = @"^(?:5[0678]\\d{0,2}|6304|67\\d{0,2})\\d{0,12}";
+static NSString *const kRegexAmex = @"^3[47]\\d{0,13}";
+static NSString *const kRegexDiscover = @"^(?:6011|65\\d{0,2}|64[4-9]\\d?)\\d{0,12}";
+static NSString *const kRegexDinersClub = @"^3(?:0([0-5]|9)|[689]\\d?)\\d{0,11}";
+static NSString *const kRegexJCB = @"^(35[2-8][0-9]).*?";
```
Comment (stpopa) [inline Source/Models/Constants/JPConstants.h:42]: please, use `^(?:35\\d{0,2})\\d{0,12}`

**Pull request merged**